internal/controller/replication.storage/volumegroupreplication_controller.go

```diff
@@ -492,36 +491,19 @@ func (r *VolumeGroupReplicationReconciler) setGroupReplicationFailure(
 func (r *VolumeGroupReplicationReconciler) getMatchingPVCsFromSource(instance *replicationv1alpha1.VolumeGroupReplication) ([]corev1.PersistentVolumeClaim, string, error) {
 
 	pvcList := corev1.PersistentVolumeClaimList{}
-	newSelector := labels.NewSelector()
 
-	if instance.Spec.Source.Selector.MatchLabels != nil {
-		for key, value := range instance.Spec.Source.Selector.MatchLabels {
-			req, err := labels.NewRequirement(key, selection.Equals, []string{value})
-			if err != nil {
-				r.log.Error(err, "failed to add label selector requirement", "Selector Key", key, "Selector Value", value)
-				return nil, "", err
-			}
-			newSelector = newSelector.Add(*req)
-		}
+	selector, err := metav1.LabelSelectorAsSelector(instance.Spec.Source.Selector)
```

Comment:
Is nil-ness check handled internally?

Reply:
Yup
Check the issue description for function definition.